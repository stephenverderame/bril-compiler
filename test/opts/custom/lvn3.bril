@f(x: int) : int {
    y: int = add x x;
    z: int = add x x;
    y: int = const 10;
    z: int = const 20;
    w: int = add x x;
    r: int = add y z;
    r: int = add r w;
    ret r;
}

@main {
    a: int = const 10;
    r: int = call @f a;
    print r;

    b: int = const 2;
    zero: int = const 0;
    r: int = mul r b;
    g: int = add r zero;
    print g;
    z: int = mul g zero;
    print z;
}