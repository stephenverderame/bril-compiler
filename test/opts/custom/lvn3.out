50
100
0
